addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -574,7 +578,7 @@ def _track_template(self, changes):
         applicant = self[0]
         # When applcant is unarchived, they are put back to the default stage automatically. In this case,
         # don't post automated message related to the stage change.
-        if 'stage_id' in changes and applicant.exists() and applicant.stage_id.template_id and not applicant._context.get('just_unarchived'):
+        if 'stage_id' in changes and applicant.exists() and applicant.stage_id.template_id and applicant._context.get('just_created'):
```

Comment:
Don't remove `just_unarchived`..  if you do, if we will spam applicants that we just unarchived.. and that's what we don't want

related comment of mine -https://github.com/odoo/odoo/pull/147559#discussion_r1568366031

Instead, you should probably have context when you change the pipe for the applicant 

`To my understanding, what was asked is to not send this mail, for the applicants that we moved to a pipe. This is because we don't want to spam people.`

relevant spec
![image](https://github.com/odoo/odoo/assets/11148685/29a3bd7d-b03b-457a-9c7f-a5ecc3722517)

Reply:
@Bertrand2  @sofiagvaladze  Apologies :pray:
I don't know what I was thinking at that moment :disappointed: 
I've fixed the code
Now it won't send the mail to those applicants